src/main/java/com/kentington/thaumichorizons/common/blocks/BlockPortalTH.java

```diff
@@ -95,29 +96,37 @@ public void onEntityCollidedWithBlock(final World world, final int x, final int
                         targetX = PocketPlaneData.planes.get(planeNum).portalA[0];
                         targetY = PocketPlaneData.planes.get(planeNum).portalA[1] - 2;
                         targetZ = PocketPlaneData.planes.get(planeNum).portalA[2];
+                        if (PocketPlaneData.planes.get(planeNum).portalA.length > 3)
+                            targetDim = PocketPlaneData.planes.get(planeNum).portalA[3];
                         break;
                     }
                     case 2: {
                         targetX = PocketPlaneData.planes.get(planeNum).portalB[0];
                         targetY = PocketPlaneData.planes.get(planeNum).portalB[1] - 2;
                         targetZ = PocketPlaneData.planes.get(planeNum).portalB[2];
+                        if (PocketPlaneData.planes.get(planeNum).portalB.length > 3)
+                            targetDim = PocketPlaneData.planes.get(planeNum).portalB[3];
                         break;
                     }
                     case 1: {
                         targetX = PocketPlaneData.planes.get(planeNum).portalC[0];
                         targetY = PocketPlaneData.planes.get(planeNum).portalC[1] - 2;
                         targetZ = PocketPlaneData.planes.get(planeNum).portalC[2];
+                        if (PocketPlaneData.planes.get(planeNum).portalC.length > 3)
+                            targetDim = PocketPlaneData.planes.get(planeNum).portalC[3];
                         break;
                     }
                     case 3: {
                         targetX = PocketPlaneData.planes.get(planeNum).portalD[0];
                         targetY = PocketPlaneData.planes.get(planeNum).portalD[1] - 2;
                         targetZ = PocketPlaneData.planes.get(planeNum).portalD[2];
+                        if (PocketPlaneData.planes.get(planeNum).portalD.length == 4)
```

Comment:
Not that it really matters but why it is ==4 here and <3 in other places? Also if you got to it, why not compact that weird switch?

Reply:
Ah whoops yeah, missed that, thanks.

And yeah I guess I'll compact it, I'll make another commit in a bit.